### ISSUE TITLE
Auto-setting defaults

### ISSUE THREAD

**Issue by tbekolay** [nengo/nengo-loihi#73]
The nengo tests and benchmarks would likely see improvements if we automatically call `nengo_loihi.set_defaults` before creating models. Should we do this? It is straightforward to do this in pytest, and I presume we can do something similar with benchmarking. But it could be considered magical when the model is either knowingly or unknowingly relying on the Nengo default.

Note that this is a separate issue from #69; the intercepts should be limited regardless of what the default or user setting is for intercepts. With `set_defaults`, we only change the default values, so any values specified by the user will be used as is.

**Comment by tcstewar** [comment]
> The nengo tests and benchmarks would likely see improvements if we automatically call `nengo_loihi.set_defaults` before creating models.

I'm not entirely sure that's true any more.  #69 greatly improves things when using the normal defaults, and `nengo.loihi.set_defaults` is pretty aggressive with its intercept setting (-0.5 to 0.5), and I'm not sure we've explored how much that affects things.  And #49 made things much better for high-firing neurons, so we might not need to reduce the firing rates to (100, 120).  In fact, reducing the firing rates that much might also cause tests and benchmarks to get worse.  

I'll try running a few benchmarks with and without set_defaults and see what the results look like....  :)

**Comment by hunse** [comment]
Yeah, let's definitely explore the effects of setting vs not setting the defaults. We can also change the defaults if we want. (My choices of -0.5 to 0.5 for the intercepts and 100 to 120 for the max rates were both pretty arbitrary. They were values that seemed to work well for early models, but a lot has changed since then, and I didn't do any parameter sweeps or anything.)

With regards to setting the defaults automatically, though, I'm not sure how we would implement this. When we're creating the model, we don't know what simulator it's going to be run on. We could see if `nengo_loihi` is imported, but then that'll give different results if the user imports it at the top (proper) vs when they create the simulator (which I sometimes do when hacking). And once the model is created, I don't think we have any way to know what was a default versus what was not (other than checking if the set value equals the default, but if we did that, then there would be no way for the user to say e.g. "I really want the intercepts to be Uniform(-1, 1), even on nengo_loihi").
